### PR TITLE
change required python version

### LINKED ISSUE #274: Reduce python 3 required version
The Darwin Python module uses Python 3.5.6. Can we please reduce the required Python version to that? I'm confident all required capabilities will be present. Pinging @JoshuaSBrown

### PR DESCRIPTION
**Pull request by Yurlungur**
## PR Summary

Reduce required version to python 3.5 instead of python 3.6. Runs on my laptop and Darwin. Resolves issue #274 where I found that Darwin used Python 3.5 by default. The differences between Python 3.5 and 3.6 are extremely minor. 

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

**Review by JoshuaSBrown — APPROVED** [review]
Looks good to me, don't know why the ci failed on the coverage test, might have been a glitch in the server.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@
 - [[PR 271]](https://github.com/lanl/parthenon/issues/256): Fix setting default CXX standard.
 - [[PR 262]](https://github.com/lanl/parthenon/pull/262) Fix setting of "coverage" label in testing. Automatically applies coverage tag to all tests not containing "performance" label.
 
+### Changed (changing behavior/API/variables/...)
+- [[PR 276]](https://github.com/lanl/parthenon/pull/276) Decrease required Python version from 3.6 to 3.5.
```
Comment (pgrete) [inline CHANGELOG.md:16]: I suggest to move it to the "Fixed (not changing behavior/API/variables/...)" category as no downstream code changes are required.

Comment (Yurlungur) [reply CHANGELOG.md:16]: Done.

**Comment by Yurlungur** [comment]
@pgrete I think your comments have been addressed.

**Pull request merged**